packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3082,6 +3087,7 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     if (_hasFocus) {
       _openInputConnection();
     } else {
+      _nextFocusChangeIsInternal = true;
```

Comment:
It's unfortunate that we have to set this boolean to true in multiple places. Maybe extract these two lines into a method?
```dart
void requestFocusInternal() {
  _nextFocusChangeIsInternal = true;
  widget.focusNode.requestFocus();
}
```
I'm not sure if this makes things better or not. What do you think?

Reply:
Sounds good, I've pulled this out into a method like this while also catching Greg's corner case from https://github.com/flutter/flutter/pull/119583#discussion_r1091314073.  It's not quite as clean as your example because I also had to support autofocus: `FocusScope.of(context).autofocus(widget.focusNode);` so I couldn't abstract it all as `widget.focusNode.requestFocus();`.